inc/composer/class-command.php

```diff
@@ -687,6 +693,192 @@ protected function import_uploads() {
+				$size = $response->getHeaderLine( 'Content-Length' );
+				$download_progress_bar->setFormat( 'Downloading ' . $this->human_filesize( $size ) . ' [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s%' );
+				$download_progress_bar->start();
+
+			},
+			'progress' => function ( int $total_bytes, int $bytes_so_far ) use ( $download_progress_bar ) {
+				if ( ! $total_bytes || ! $bytes_so_far ) {
+					return;
+				}
+				$download_progress_bar->setProgress( $bytes_so_far / $total_bytes * 100 );
+			},
+		] );
+
+		$download_progress_bar->finish();
+		$output->writeln( 'Complete' );
+
+		$extract_dir = sys_get_temp_dir() . '/export';
+		if ( is_dir( $extract_dir ) ) {
+			$this->rmdir_recursive( $extract_dir );
+		}
+		try {
+			$phar = new PharData( $download_file_path );
+			$phar->extractTo( $extract_dir ); // extract all files
+		} catch ( Exception $e ) {
+			$output->writeln( sprintf( '<error>Unable to extract tar file: %s</error>', $e->getMessage() ) );
+			// To do: cleanup
+			return;
+		}
+
+		if ( file_exists( $extract_dir . '/database.sql.gz' ) ) {
+			$output->writeln( 'Database found in export, importing...' );
+			$database_file_path = 'database.sql';
+			$sql_file = $this->gunzip_file( $extract_dir . '/database.sql.gz', $database_file_path );
+
+			// Import the file into mysql
+			$cli = $this->getApplication()->find( 'local-server' );
+			$import = $cli->run( new ArrayInput( [
+				'subcommand' => 'cli',
+				'options' => [
+					'db',
+					'import',
+					$sql_file
+				],
+			] ), $output );
+
+			// Flush cache
+			$flush_cache = $cli->run( new ArrayInput( [
+				'subcommand' => 'cli',
+				'options' => [
+					'cache',
+					'flush'
+				],
+			] ), $output );
+
+
+			ob_start();
+			$export_sites = $cli->run( new ArrayInput( [
+				'subcommand' => 'cli',
+				'options' => [
+					'db',
+					'query',
+					'SELECT DISTINCT domain from wp_site'
+				],
+			] ), $output );
+			$sites_output = ob_get_clean();
+			// Oh lordy! There's no good way to run a query via WP-CLI and get machine readable output.
+			preg_match_all( '/ (\S+) /', $sites_output, $domains, PREG_SET_ORDER );
+			$domains = array_map( function ( array $match ) : string {
+				return $match[1];
+			}, $domains );
+			$domains = array_diff( $domains, [ 'domain' ] );
+
+			$output->writeln( sprintf( 'Replacing URLs for sites %s.', implode( ', ', $domains ) ) );
+			$replacement_domain = $this->get_project_subdomain() . '.altis.dev';
+			foreach ( $domains as $export_domain ) {
+				$cli->run( new ArrayInput( [
+					'subcommand' => 'cli',
+					'options' => [
+						'search-replace',
+						$export_domain,
+						$replacement_domain,
+						'--network',
+						'--url=' . $export_domain,
+					],
+				] ), $output );
+			}
+
+			// Flush cache
+			$flush_cache = $cli->run( new ArrayInput( [
+				'subcommand' => 'cli',
+				'options' => [
+					'cache',
+					'flush'
+				],
+			] ), $output );
+
+			$elasticpress_index = $cli->run( new ArrayInput( [
```

Comment:
We also have the `wp altis migrate` command now but that's technically not required, just wondering if we can trigger everything hooked onto the `multisite-install` command instead, without actually running that command!